src/NuGet.Core/NuGet.Protocol/Model/IPackageSearchMetadata.cs

```diff
@@ -37,6 +37,8 @@ public interface IPackageSearchMetadata
         bool IsListed { get; }
         bool PrefixReserved { get; }
 
+        bool? HasReadme { get; }
```

Comment:
Is this `bool` property going to look short-sighted once readme support for HTTP URLs is added?

As in another comment, some HTTP feeds will signal there is a readme by having `ReadmeUrl` not null, and a particular resource in the service index set.  For those feeds, this `bool HasReadme` is not needed.

Other HTTP feeds will need a new `Uri RawReadmeUrl` (or maybe `string`). This bool won't be sufficient, and therefore this `IPackageSearchMetadata` will need to be modified again, at which time it will have a `ReadmeUrl`, `RawReadmeUrl`, and also `HasReadme` properties.  Except `HasReadme` is computed from other properties (and the service index), it's not DTO data with a JSON property in the protocol.

For local feeds, when the package is extracted, the readme file full path could be put in the `RawReadmeUrl`. For feeds that are just a directory full of nupkgs, the nupkg's full path could be provided in the `RawReadmeUrl`.

So, I think that this bool property is unnecessary, and by adding a `RawReadmeUrl` instead, we won't need to make more public API changes when readme for HTTP feeds is added later.

Reply:
hmm yea that sounds reasonable. I've already removed this bool.